ospray_modules/optix/Context.cpp

```diff
@@ -283,50 +248,155 @@ void Context::destroy()
-    rtDeviceGetDeviceCount(&num_devices);
-    rtGetVersion(&version);
+    _phong_ch =
+        _optixContext->createProgramFromPTXString(CUDA_PHONG,
+                                                  "closest_hit_radiance");
+
+    _phong_ch_textured = _optixContext->createProgramFromPTXString(
+        CUDA_PHONG, "closest_hit_radiance_textured");
+
+    _phong_ah =
+        _optixContext->createProgramFromPTXString(CUDA_PHONG, "any_hit_shadow");
+
+    _bounds[Geometry::Type::Cones] =
+        _optixContext->createProgramFromPTXString(CUDA_CONES,
+                                                  CUDA_FUNCTION_BOUNDS);
+    _intersects[Geometry::Type::Cones] =
+        _optixContext->createProgramFromPTXString(CUDA_CONES,
+                                                  CUDA_FUNCTION_INTERSECTION);
 
-    ospray::postStatusMsg()
-        << "Number of CUDA Devices: " + std::to_string(num_devices);
+    _bounds[Geometry::Type::Cylinders] =
+        _optixContext->createProgramFromPTXString(CUDA_CYLINDERS,
+                                                  CUDA_FUNCTION_BOUNDS);
+    _intersects[Geometry::Type::Cylinders] =
+        _optixContext->createProgramFromPTXString(CUDA_CYLINDERS,
+                                                  CUDA_FUNCTION_INTERSECTION);
 
-    for (unsigned int i = 0; i < num_devices; ++i)
+    _bounds[Geometry::Type::Spheres] =
+        _optixContext->createProgramFromPTXString(CUDA_SPHERES,
+                                                  CUDA_FUNCTION_BOUNDS);
+    _intersects[Geometry::Type::Spheres] =
+        _optixContext->createProgramFromPTXString(CUDA_SPHERES,
+                                                  CUDA_FUNCTION_INTERSECTION);
+
+    _bounds[Geometry::Type::TriangleMesh] =
+        _optixContext->createProgramFromPTXString(CUDA_TRIANGLES_MESH,
+                                                  CUDA_FUNCTION_BOUNDS);
+    _intersects[Geometry::Type::TriangleMesh] =
+        _optixContext->createProgramFromPTXString(CUDA_TRIANGLES_MESH,
+                                                  CUDA_FUNCTION_INTERSECTION);
+}
+
+void Context::_printSystemInformation() const
+{
+    unsigned int optixVersion;
+    RT_CHECK_ERROR_NO_CONTEXT(rtGetVersion(&optixVersion));
+
+    unsigned int major = optixVersion / 1000; // Check major with old formula.
+    unsigned int minor;
+    unsigned int micro;
+    if (3 < major) // New encoding since OptiX 4.0.0 to get two digits micro
+                   // numbers?
+    {
+        major = optixVersion / 10000;
+        minor = (optixVersion % 10000) / 100;
+        micro = optixVersion % 100;
+    }
+    else // Old encoding with only one digit for the micro number.
     {
-        char deviceName[256];
-        int computeCaps[2];
-        int clock_rate;
-        uint64_t totalMemory{0};
+        minor = (optixVersion % 1000) / 10;
+        micro = optixVersion % 10;
+    }
+    ospray::postStatusMsg() << "OptiX " << major << "." << minor << "."
```

Comment:
Can postStatusMsg be used without initializing OSPray, or is it always done anyway?

Reply:
This is dependent on OSPRay but since the optix plugin is actually a plugin that is loaded by OSPRay it should always be set.